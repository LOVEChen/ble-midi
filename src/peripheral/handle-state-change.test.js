/* eslint-env jest */
jest.mock('../util/logger');

let handleStateChange;
let mockLog;

beforeEach(() => {
  jest.resetAllMocks();
  mockLog = require('../util/logger').default;
  handleStateChange = require('./handle-state-change').default;
});

test('should handleStateChange when poweredOn', () => {
  const bleno = jest.fn();
  bleno.startAdvertising = jest.fn();
  handleStateChange(bleno, 'mock-name', 'poweredOn');
  expect(bleno.mock).toMatchSnapshot();
  expect(bleno.startAdvertising.mock).toMatchSnapshot();
  expect(mockLog.mock.calls).toMatchSnapshot();
});

test('should handleStateChange when not poweredOn', () => {
  const bleno = jest.fn();
  bleno.stopAdvertising = jest.fn();
  handleStateChange(bleno, 'mock-name', 'mock-state');
  expect(bleno.mock).toMatchSnapshot();
  expect(bleno.stopAdvertising.mock).toMatchSnapshot();
  expect(mockLog.mock.calls).toMatchSnapshot();
});
