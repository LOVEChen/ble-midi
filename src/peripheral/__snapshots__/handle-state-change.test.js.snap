// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handleStateChange when not poweredOn 1`] = `
Object {
  "calls": Array [],
  "instances": Array [],
}
`;

exports[`should handleStateChange when not poweredOn 2`] = `
Object {
  "calls": Array [
    Array [],
  ],
  "instances": Array [
    [Function],
  ],
}
`;

exports[`should handleStateChange when not poweredOn 3`] = `
Object {
  "calls": Array [
    Array [
      "stateChange: mock-state",
    ],
    Array [
      "Stop advertising Peripheral \\"mock-name\\": 03B80E5A-EDE8-4B33-A751-6CE34EC4C700",
    ],
  ],
  "instances": Array [
    undefined,
    undefined,
  ],
}
`;

exports[`should handleStateChange when poweredOn 1`] = `
Object {
  "calls": Array [],
  "instances": Array [],
}
`;

exports[`should handleStateChange when poweredOn 2`] = `
Object {
  "calls": Array [
    Array [
      "mock-name",
      Array [
        "03B80E5A-EDE8-4B33-A751-6CE34EC4C700",
      ],
    ],
  ],
  "instances": Array [
    [Function],
  ],
}
`;

exports[`should handleStateChange when poweredOn 3`] = `
Object {
  "calls": Array [
    Array [
      "stateChange: poweredOn",
    ],
    Array [
      "Start Advertising Peripheral \\"mock-name\\": 03B80E5A-EDE8-4B33-A751-6CE34EC4C700",
    ],
  ],
  "instances": Array [
    undefined,
    undefined,
  ],
}
`;
